### PR TITLE
Adsk Contrib - Make Imath dependency optional

### PR DESCRIPTION
**Pull request by cozdas**
- Adding a slightly modified version of Imath half header files in header-only form into our repo.

- OCIO_USE_HALF_LOOKUP_TABLE is set, then we'll use the external full Imath library, but if it's not set then we'll switch to using this internal header-only implementation, reducing the external dependency of OCIO.

- in few places we were including the half.h using unnecessary parent path as "Imath/half.h". This was working as the parent directory was also in the header search paths but was unnecessary and misleading as the path to the half.h directly was also in the search path. Removing this extra level made the switching between the externa/internal half implementation smooth and clear.

### CODE REVIEW

File: ext/half/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+There are only few modifications on the original code:
+- turned off the LUT by forcing the IMATH_HALF_NO_LOOKUP_TABLE definition 
+- commented out std::ostream& >> and << operators
+- commented out printBits() functions.
+
+Please note that except for the LUT and test routines, the half support in OCIO is utilizing the 
+hardware conversion if the library is built with F16C suport turned on (default) and runnin on a CPU
+that has the F16C instruction set. Therefore the performance impact of not-using the half lookup table
+is very limited on typical cases.
```
Comment (doug-walker) [inline ext/half/README.md:18]: runnin --> running

I want to understand what happens when using our half.h on CPUs that either support or not the hardware instructions. Are you saying that we are switching dynamically at runtime to support either case?  Or that the app has to decide which one it wants to support and there is no runtime switch?

**Review by doug-walker — COMMENTED** [review]
This looks great!  I want to understand when this is active.  The UseLut option is the default, so this would not by default be active in full OCIO.

However, in this branch nano mode is the default and nano turns off UseLut, correct?  So for all of the CI tests that are running, they are running in nano mode?

I notice that there are some failures with OIIO looking for Imath and not finding it. I think if we are in nano mode we should turn OIIO mode off.

File: ext/half/README.md (hunk 1 above)
Comment (cozdas) [reply ext/half/README.md:18]: Our half.h is in use only when the OCIO_USE_HALF_LOOKUP_TABLE is OFF.  This flag is ON in the Full mode, and OFF in the nano mode. So our half.h is used only in the Nano mode (or if OCIO_USE_HALF_LOOKUP_TABLE manually turned off in the Custom mode).

My current understanding is that we need half<->float conversion mainly in two parts of the OCIO library:
1- LUTs: Lut1D for sure, probably Lut3D too.
2- Pixels: scanline helpers

1- Lut code is always using whatever the Imath provides: 64k lookup table, bit shifting or hardware conversion (if compiled with F16C flag, in our case it's not, as we want to support older CPUs too)

2- Scanline helpers detect the CPU capability at run-time and handling the half numbers with hardware SSE registers while processing multiple channels at the same time. So if OCIO is compiled with proper SSE or AVX support (default on) and if the CPU has the SSE instructions available, then for scanline processing OCIO doesn't need Imath library at all. Scanline helpers will use imath conversion routines only when our hand-coded parallel half handling routines can't be used. Let me know if my understanding is incorrect or incomplete. 

All that means is, when the OCIO_USE_HALF_LOOKUP_TABLE is turned off, there is no functional difference between using the external Imath dependency and using our internal header-only half.h. I'm trying to explain the impact of turning off the OCIO_USE_HALF_LOOKUP_TABLE in that paragraph. Suggestions to make it clearer is welcome.

Comment (doug-walker) [reply ext/half/README.md:18]: Thanks for the comments. What you wrote is not matching how I thought it worked. But let's discuss live next week.

File: CMakeLists.txt [hunk 2]
```diff
@@ -487,27 +487,28 @@ elseif(NOT OCIO_FEATURE_SET MATCHES "[Cc][Uu][Ss][Tt][Oo][Mm]")
...
+    message(WARNING "Documentation requires Python bindings which is turned off, thus OCIO_BUILD_DOCS option will be turned off too.")
     set(OCIO_BUILD_DOCS OFF CACHE BOOL "Specify whether to build documentation" FORCE)
 endif()
 
-if(OCIO_FEATURE_SET MATCHES "[Nn][Aa][Nn][Oo]" AND OCIO_BUILD_OPENFX)
-    message(WARNING "OpenFx is not compatible with Nano feature set,\
-    thus OCIO_BUILD_OPENFX option will be turned off.")
-    set(OCIO_BUILD_OPENFX OFF CACHE BOOL "Specify whether to build OpenFX plugins" FORCE)
+if(OCIO_USE_OIIO_FOR_APPS AND NOT OCIO_USE_HALF_LOOKUP_TABLE)
+	message(WARNING "OIIO is needs OCIO_USE_HALF_LOOKUP_TABLE which is turned off, thus OCIO_USE_OIIO_FOR_APPS option will be turned off too.")
```
Comment (doug-walker) [inline CMakeLists.txt:505]: "is needs" --> "needs"

**Review by doug-walker — APPROVED** [review]

**Pull request merged**